crates/bevy_text/src/text.rs

```diff
@@ -174,20 +174,69 @@ pub struct TextStyle {
     ///
     /// A new font atlas is generated for every combination of font handle and scaled font size
     /// which can have a strong performance impact.
-    pub font_size: f32,
+    pub font_size: FontSize,
     pub color: Color,
 }
 
 impl Default for TextStyle {
     fn default() -> Self {
         Self {
             font: DEFAULT_FONT_HANDLE.typed(),
-            font_size: 12.0,
+            font_size: FontSize::Default,
             color: Color::WHITE,
         }
     }
 }
 
+/// The default font size.
+/// Used when the size field of TextStyle is set to Default.
+#[derive(Resource, Clone, Copy, Debug)]
+pub enum DefaultFontSize {
```

Comment:
Could this instead be a new-type over `FontSize`, with error handling to warn about a recursive definition? It would remove a few lines worth of repetition.